### ISSUE TITLE
Heroku Button (for demo) Fails on Build

### ISSUE THREAD

**Issue by graygilmore** [solidusio/solidus#1299]
The Heroku button in the Readme is currently broken.

```bash
-----> Fetching set buildpack https://github.com/solidusio/heroku-buildpack-solidus-demo.git... done
-----> Ruby app detected
-----> Compiling Ruby/Rails
       Compiling solidus
### SNIPPED CODE
20 gems installed
/app/vendor/ruby-2.3.1/lib/ruby/2.3.0/rubygems/dependency.rb:319:in `to_specs': Could not find 'railties' (~> 4.2.0) - did find: [railties-5.0.0] (Gem::LoadError)
Checked in 'GEM_PATH=/app/vendor/bundle/ruby/2.3.0:/app/vendor/ruby-2.3.1/lib/ruby/gems/2.3.0', execute `gem env` for more information
    from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/rubygems/dependency.rb:328:in `to_spec'
    from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_gem.rb:65:in `gem'
    from -e:1:in `<main>'
logger: unrecognized option '--skip-bundle'
### SNIPPED CODE
 !
 !     " new sandbox --skip-bundle --database=postgresql" failed
 !
/tmp/buildpack20160705-159-ts13py/lib/language_pack/solidus.rb:88:in `sh': " new sandbox --skip-bundle --database=postgresql" failed (RuntimeError)
    from /tmp/buildpack20160705-159-ts13py/lib/language_pack/solidus.rb:29:in `compile'
### SNIPPED TRACE
 !     Push rejected, failed to compile Ruby app
```

**Comment by jasonfb** [comment]
strange, that looks like a rubygems.org problem. Does ```bundle install``` on the very same app work?

**Comment by jasonfb** [comment]
there definitely is a rubygem for railties at the exact version ```4.2.0```.  are you sure your Gemfile has specified rubygems.org as the source? 

I see now you wrote "Heroku Button". I don't really know what that does exactly and Solidus is likely too complicated to deploy with button. I'd try to make yourself a brand new app and see if you can re-create the problem on it.

**Comment by graygilmore** [comment]
@jasonfb just a bit of background, I work at Stembolt 😄 

This is in reference to the `Deploy with Heroku` button on the main page of [this repo](https://github.com/solidusio/solidus#demo).

**Comment by jasonfb** [comment]
Oh ok, got it now. My opinion is that button should just go. Deploying with a button isn't such a great idea and is not something that is going to be consistent and easy to support. (I don't know why Heroku encourages it, it's really not a such good idea). Seems like it breaks often, and Heroku's build procedure changes too, and moving forward this button is just a headache. 

We maintain a fresh install app ("hello world") app here: http://solidus-1-3-helloworld.herokuapp.com

I can modify this helloworld app with the latest released version and also make a new one for 1.4 now.

**Comment by graygilmore** [comment]
As far as I know the idea behind the button is just to try it out, not to actually start work on a new project.

**Comment by jasonfb** [comment]
Yes I hear ya, it's just that this is the umpteenth time I've seen people have problems with it and Heroku has the power to change stuff on their side and break the button (which admittedly, happens rarely, but can happen). All I'm saying is that supporting it seems untenable to me.

**Comment by marknfischer** [comment]
I had same issue last week and today with this button.  Would love to "try it out with the click of a button."  I'm obviously not going into production with that architecture.  But seeing it working w/o spinning up a local rails instance would be awesome.

**Comment by cassidyclawson** [comment]
Just another +1 that the button doesn't work. I think ya'll show try to save it because the button is a great way for people like me to demo the app.

**Comment by dfranciosi** [comment]
I can confirm that it doesn't work yet
` could not connect to server: No such file or directory (PG::ConnectionBad)
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
`

**Comment by michaelminter** [comment]
I get 

```
Bundle completed (39.73s)
       Cleaning up the bundler cache.
 !
 !     cannot load such file -- bundler/lockfile_parser
 !
/app/vendor/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- bundler/lockfile_parser (LoadError)
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/helpers/bundler_wrapper.rb:128:in `block in parse_gemfile_lock'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:16:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/helpers/bundler_wrapper.rb:85:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/helpers/bundler_wrapper.rb:126:in `parse_gemfile_lock'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/helpers/bundler_wrapper.rb:110:in `lockfile_parser'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/helpers/bundler_wrapper.rb:73:in `specs'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/helpers/bundler_wrapper.rb:53:in `has_gem?'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/ruby.rb:659:in `block in create_database_yml'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:16:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:48:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:44:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/ruby.rb:657:in `create_database_yml'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/ruby.rb:102:in `block (2 levels) in compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/ruby.rb:751:in `allow_git'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/ruby.rb:98:in `block in compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:16:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:48:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:44:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/ruby.rb:88:in `compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/rails2.rb:49:in `block in compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:16:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:48:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:44:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/rails2.rb:47:in `compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/rails3.rb:38:in `block in compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:16:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:48:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:44:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/rails3.rb:37:in `compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/rails4.rb:41:in `block in compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:16:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:48:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:44:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/rails4.rb:40:in `compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/solidus.rb:66:in `compile'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/bin/compile:16:in `block (2 levels) in <main>'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/base.rb:131:in `log'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/bin/compile:15:in `block in <main>'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:35:in `block in trace'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:17:in `block in instrument'
	from /app/vendor/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:16:in `instrument'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/lib/language_pack/instrument.rb:35:in `trace'
	from /app/tmp/buildpacks/4503e3a76977915f47253edc22ff766cf41874cec155814eeb2bed137037cc920729733d948ffeb2124e6612e9bfcc160ea3849de67fe7190e0f7d126454c288/bin/compile:11:in `<main>'
 !     Push rejected, failed to compile Ruby app.
 !     Push failed
```

**Comment by sheerun** [comment]
Confirming

**Comment by nhodges** [comment]
Now it fails because of interdependency issues.

Gemfile requires `rails (= 4.2.7.1)` however the version of solidus_core it resolves to requires `rails (~> 5.0.0)`

**Comment by jhawthorn** [comment]
This should be [fixed](https://test-solidus-rails-5.herokuapp.com/)!

I can't thank @stewart enough for fixing the issues with bundler and updating the buildpack to rails 5.
